ibis/pyspark/compiler.py

```diff
@@ -102,15 +106,25 @@ def compile_sql_query_result(t, expr, scope, timecontext, **kwargs):
 @compiles(ops.Selection)
 def compile_selection(t, expr, scope, timecontext, **kwargs):
     op = expr.op()
-
-    src_table = t.translate(op.table, scope, timecontext)
+    # In selection, there could be multiple children that point to the
+    # same root table. e.g. window with different sizes on a table.
+    # We need to get the 'max' time range that is a superset of every
```

Comment:
I'd prefer to change "max" to combined. Max has a well defined meaning.

Reply:
+1